src/Import/src/FMI3/fmi3_import.c

```diff
@@ -102,7 +102,9 @@ fmi3_import_t* fmi3_import_parse_xml(
 
     if (!fmu) {
         context->callbacks->free(xmlPath);
-        if (!terminalsAndIconsPath) context->callbacks->free(terminalsAndIconsPath);
+        if (terminalsAndIconsPath) {
```

Comment:
Would it make sense to wrap this `if (terminalsAndIconsPath) { free ... }` in one function?

Reply:
I don't think so, there is only 2 uses.

There is some code duplication in this function to handle the cleanup in the various possible failure stages.
The corresponding fmi2 function does handle it with less code, but uses goto ... so I do prefer this alternative.